torch/csrc/jit/runtime/autodiff.cpp

```diff
@@ -356,6 +356,28 @@ static Value* createAutogradAdd(Value* a, Value* b) {
   return graph->insertNode(graph->create(prim::AutogradAdd, {a, b}))->output();
 }
 
+namespace {
+bool outputRequiresGrad(Value* output) {
+  if (output->type()->castRaw<TensorType>() == nullptr) {
+    return output->requires_grad();
```

Comment:
dont know when we would be putting lists and such to have grad here but is consistent with existing behavior so looks good

Reply:
for lists we don't profile so I think we have to assume that they require_grad